lib/model/content.dart

```diff
@@ -1093,10 +1110,32 @@ class _ZulipContentParser {
     } else {
       return UnimplementedBlockContentNode(htmlNode: divElement);
     }
+
+    double? originalWidth, originalHeight;
+    final originalDimensions = imgElement.attributes['data-original-dimensions'];
+    if (originalDimensions != null) {
+      // Server encodes this string as "{width}x{height}" (eg. "300x400")
+      final match = _imageDimensionsRegExp.firstMatch(originalDimensions);
+      if (match != null) {
+        final width = int.tryParse(match.group(1)!);
+        final height = int.tryParse(match.group(2)!);
```

Comment:
There's a subtle and unfortunate gotcha in Dart's `int.parse` and `int.tryParse`, which it's best to explicitly disable:
```suggestion
        final width = int.tryParse(match.group(1)!, radix: 10);
        final height = int.tryParse(match.group(2)!, radix: 10);
```

For discussion, see f5477717aec34590bee82298db3a64800bb14877.

In this case I believe it doesn't have any effect on the behavior, because we already know from the regexp that each group matches exactly `\d+` and so can't trigger this gotcha. But that's a pretty subtle and nonlocal property, so it's best to avoid the gotcha more directly too: that way (a) the gotcha won't easily be reintroduced by an otherwise-innocent change elsewhere in this logic, and (b) it maintains the pattern for the benefit of anyone copy-pasting or adapting this code for somewhere else in the codebase.

Reply:
(Separately I was tempted to say `tryParse` can be just `parse`, because we already know they match exactly `\d+`. But if we got an image with something like `data-original-dimensions="1234123412341234123412341234x5"`, the parse would indeed fail because the number is too big for an int. So `tryParse` is more correct — better to return unimplemented than to throw — and it's good we have it.)